app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  after_create :increment_post_comments_counter
+
```

Comment:
---

Hello,

I've reviewed your `Comment` model and noticed that it's missing a key piece of functionality based on the project requirements. Specifically, there's a requirement for a method that updates the comments counter for a post. This method is essential for keeping track of the number of comments a post has received.

Here is the requirement that seems to be missing:

- Comment model has a method that updates the comments counter for a post.

You can implement this by defining an instance method in the `Comment` model that updates the associated post's `comments_counter` attribute. Here's a basic example of how this method could be implemented:

```ruby
class Comment < ApplicationRecord
  belongs_to :post

  # Other code for the Comment model

  # Method to update the post's comments counter
  def update_comments_counter
    post.increment!(:comments_counter)
  end
end
```

The `increment!` method is a convenient way to increase the counter by one, and it automatically saves the record, which is what we want in this case.

Remember to call this method in the appropriate place, typically after a comment is successfully created or destroyed, to ensure the counter remains accurate.

Please implement this method and ensure it is being called at the correct times. If you have any questions about this process or run into any difficulties, don't hesitate to ask for help.

Keep up the good work, and I look forward to seeing your revised code!

Best regards,

Temitope

---